### ISSUE TITLE
NSOpenGL Deprecation warnings/errors on MacOS 10.14.6 and above

### ISSUE THREAD

**Issue by dbsmythe** [materialx/MaterialX#327]
When trying to compile "source/MaterialXRenderGlsl/CMakeFiles/MaterialXRenderGlsl.dir/GLCocoaWrappers.m.o" on MacOS 10.14.6 (or higher, presumably), I get a large number of deprecation warnings such as the following:

```
/Users/smythe/Documents/GitHub/MaterialX/source/MaterialXRenderGlsl/GLCocoaWrappers.m:19:5: warning: 
      'NSOpenGLPixelFormatAttribute' is deprecated: first deprecated in macOS 10.14 - OpenGL API
      deprecated; please use Metal and MetalKit. (Define GL_SILENCE_DEPRECATION to silence these
      warnings.) [-Wdeprecated-declarations]
    NSOpenGLPixelFormatAttribute list[50];
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:101:18: note: 
      'NSOpenGLPixelFormatAttribute' has been explicitly marked deprecated here
typedef uint32_t NSOpenGLPixelFormatAttribute NS_OPENGL_DEPRECATED(10.0, 10.14);
                 ^
/Users/smythe/Documents/GitHub/MaterialX/source/MaterialXRenderGlsl/GLCocoaWrappers.m:23:21: warning: 
      'NSOpenGLPFAAllRenderers' is deprecated: first deprecated in macOS 10.14 - OpenGL API
      deprecated; please use Metal and MetalKit. (Define GL_SILENCE_DEPRECATION to silence these
      warnings.) [-Wdeprecated-declarations]
        list[i++] = NSOpenGLPFAAllRenderers;
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:60:5: note: 
      'NSOpenGLPFAAllRenderers' has been explicitly marked deprecated here
    NSOpenGLPFAAllRenderers          NS_OPENGL_ENUM_DEPRECATED(10.0, 10.14)  =   1,     /...
    ^
```
as well as the following ERROR:
```
/Users/smythe/Documents/GitHub/MaterialX/source/MaterialXRenderGlsl/GLCocoaWrappers.m:202:5: error: 
      unknown type name 'NSOpenGLContextAuxiliary'
    NSOpenGLContextAuxiliary* contextAuxiliary =  [context CGLContextObj];
    ^
```
As a result, MaterialXRender can no longer be compiled on MacOS, and unfortunately, just turning off MATERIALX_BUILD_VIEWER in CMake is not sufficient to make the build successful.

**Comment by jstone-lucasfilm** [comment]
Thanks for the heads-up, Doug, and we should implement the following fixes:

- Integrate Ashwin's MacOS fix from the Autodesk fork (https://github.com/autodesk-forks/MaterialX/pull/647).
- Add the GL_SILENCE_DEPRECATION compiler flag to MacOS builds.
- Increment the osx_image setting for Travis builds, so that we're testing with a more recent MacOS environment.

One additional note is that Ashwin's fix is already available in the v1.37 branch of MaterialX, so it should be possible to build (with deprecation warnings) in recent MacOS environments using that branch.

**Comment by jstone-lucasfilm** [comment]
Fixed in #329
